parquet-common/src/main/java/org/apache/parquet/io/OutputFile.java

```diff
@@ -31,5 +31,7 @@
 
   long defaultBlockSize();
 
-  String getPath();
+  default String getPath() {
+    throw new UnsupportedOperationException();
```

Comment:
We're consuming this function in one of the encryption options:
https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetWriter.java#L284
If there is no encryption, or encryption is used with a crypto factory that doesn't need the file path - this exception will break the existing callers. A simple way to resolve this is by catching the exception here, and passing a null file path. This allows for these callers to run ok. The crypto factories that do need the file path sometimes (like the one we have in parquet-mr) should check `path==null` in situations where the file path is used, and throw an exception there. What do you think?

Reply:
It sounds good to me. I'll exchange the default implementation to return `null`.

Reply:
Cool, thanks. I'll add then the null checks in the crypto-factory (key management).

Reply:
I think, I've added them. Please review the latest commit.

Reply:
Oh, this is already covered in the commit. Looks good!

Reply:
+1 for using null instead of through exception, which is expensive.